### DISCUSSION TITLE
User account types (Personal, Creator or Business)

### DISCUSSION THREAD

**Discussion by uniqby** [sbertix/Swiftagram#204] [Implemented ideas] [answered]
How to detect User account type (Personal, Creator or Business)?

**Comment by sbertix** [comment]
Hey @uniqby, 

As of right now you're left to parse the underlying `Wrapper` in every `User` response. 
You can print any `user.wrapped` to see all the available parameters and info, and then call it with, for instance, `user.parameter.string()` or `user.parameter.int()`.  

I had already noticed this got lost in the various update so I'm definitely gonna push an update for this asap, hopefully before Monday.

**Comment by sbertix** [comment, marked as answer]
Hey @uniqby, 

Starting with `5.1.0` you can now detect an account type from a `User`'s `access` property as such:

```swift
// Given a `User`.
let user: User = /* some `User` */
guard let access = user.access else { return }
// Is it a business account?
switch true {
case access.contains(.business): print("It's a business account")
case access.contains(.creator): print("It's a creator account")
default: print("It's a personal account")
}
```

**Reply by uniqby** [reply]
Thank you
